### DISCUSSION TITLE
Cache not invalidated when changing browserslist configuration

### DISCUSSION THREAD

**Discussion by joliss** [vercel/next.js#43252] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: linux
      Arch: x64
      Version: #49-Ubuntu SMP Thu Aug 4 18:03:25 UTC 2022
    Binaries:
      Node: 18.7.0
      npm: 8.15.0
      Yarn: 1.22.19
      pnpm: 7.14.2
    Relevant packages:
      next: 13.0.5-canary.4
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

### Describe the Bug

When I add the following snippet to my `package.json`

```json
  "browserslist": [
    "defaults, supports es5"
  ],
```

re-running `next build && next export` does not honor the updated browserslist configuration.

Running `rm -r .next` fixes the issue, so this seems to be a cache invalidation bug.

### Expected Behavior

I shouldn't have to delete `.next`.

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code will be addressed faster

https://github.com/joliss/next-browserslist-cache-invalidation

### To Reproduce

```
$ npm install
$ npx next build && npx next export &&
    if grep -l '=>' out/_next/static/chunks/*.js; then echo ES6; else echo ES5; fi
...
ES6

$ patch -p0 < package.json.patch # set browserslist to ES5
$ npx next build && npx next export &&
    if grep -l '=>' out/_next/static/chunks/*.js; then echo ES6; else echo ES5; fi
...
ES6    <----- bug

$ rm -r .next # workaround
$ npx next build && npx next export &&
    if grep -l '=>' out/_next/static/chunks/*.js; then echo ES6; else echo ES5; fi
...
ES5
```

**Comment by balazsorban44** [comment]
Correct, the `.next` folder is cleaned [by default](https://nextjs.org/docs/upgrading#cleaning-the-distdir-is-now-a-default), but some things are cached in the `.next/cache` directory. We can add a browserslist config as a cache invalidation dependency to automatically invalidate the cache if the config changes. Will have a look.

Moving to discussion for now.


*UPDATE*:

Adding the `.browserslist` file should not be an issue, but watching `package.json` might be bad in terms of cache invalidation so this needs some extra thought.

**Reply by Noitidart** [reply]
Thanks for these replies. I have my browserlist set to this below. However after building, and then loading page, and looking at the `main-HASH.js` file, I still find `let`s and `...`s. I'm using `"next": "^13.1.6"` could you please advise how to fix this? 

package.json:

```
{
/////// stuff
  "browserslist": [
    ">0.3%",
    "defaults",
    "supports es5"
  ]
}
```

Here is my `next.config.js`:

<img width="402" alt="image" src="https://user-images.githubusercontent.com/6372489/219249225-df5ad46c-46c7-46b7-9790-83fa35ed7f44.png">


I tried clearing cash on builds too on Vercel by choosing redploy and unchecking the box to use cache:
<img width="468" alt="image" src="https://user-images.githubusercontent.com/6372489/219247486-fef1a7c3-c5d1-405a-a69e-6a157eb9f567.png">


Examples of things in the `main-HASH.js` file, these codes are from files from `next` as we see things like `normalizePathTrailingSlash`, and I did a search in `node_modules` and found it in `next` files.

<img width="1576" alt="image" src="https://user-images.githubusercontent.com/6372489/219247058-fb5ae7dd-38e8-47f2-a683-2b4f2496d1e8.png">

<img width="1580" alt="image" src="https://user-images.githubusercontent.com/6372489/219247112-f05a3655-33dd-4bd8-931a-5038a016fe53.png">

**Comment by meduzen** [comment]
Encountered the same situation when adding a package with a browserslist config. My current approach is to clear Webpack cache after each `npm install`, using a [`postinstall` script](https://docs.npmjs.com/cli/v9/using-npm/scripts#pre--post-scripts). A minimal `package.json` would look like this:

```json
{
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "export": "next export",
    "start": "next start",
    "build:export": "next build && next export",
    "postinstall": "rm -rf .next/cache/webpack"
  },
  "browserslist": [
    "extends @my_company/browserslist-config"
  ]
}

```
